client/checkout/stripe-link/index.js

```diff
@@ -4,6 +4,13 @@
 import { dispatchChangeEventFor } from '../utils/upe';
 
 export const switchToNewPaymentTokenElement = () => {
+	// Switch to card payment method before enabling new payment token element
+	document
+		.querySelector(
+			'input[name="payment_method"][value="woocommerce_payments"]'
+		)
+		?.click();
+
 	const newPaymentTokenElement = document.getElementById(
 		'wc-woocommerce_payments-payment-token-new'
```

Comment:
Once https://github.com/Automattic/woocommerce-payments/pull/9782 is merged, I think we could look new token element from within the `.wcpay-upe-form` container we created there, but since I'm unsure about PR merging sequence just yet, I avoided to set that branch as the upstream for this PR.